pkg/backend/backend.go

```diff
@@ -230,3 +233,34 @@ func (c *backendClient) GetStackOutputs(ctx context.Context, name string) (resou
 func (c *backendClient) DownloadPlugin(ctx context.Context, plug workspace.PluginInfo) (io.ReadCloser, error) {
 	return nil, errors.New("downloading plugins at runtime not available when using local backend")
 }
+
+func (c *backendClient) GetStackResourceOutputs(
+	ctx context.Context, name string) (resource.PropertyMap, error) {
+	ref, err := c.backend.ParseStackReference(name)
+	if err != nil {
+		return nil, err
+	}
+	s, err := c.backend.GetStack(ctx, ref)
+	if err != nil {
+		return nil, err
+	}
+	if s == nil {
+		return nil, errors.Errorf("unknown stack %q", name)
+	}
+	snap, err := s.Snapshot(ctx)
+	if err != nil {
+		return nil, err
+	}
+	pm := resource.PropertyMap{}
+	for _, r := range snap.Resources {
+		if r.Delete {
+			continue
+		}
+
+		resc := resource.PropertyMap{
+			resource.PropertyKey("type"):    resource.NewStringProperty(string(r.Type)),
+			resource.PropertyKey("outputs"): resource.NewObjectProperty(r.Outputs)}
+		pm[resource.PropertyKey(r.URN)] = resource.NewObjectProperty(resc)
```

Comment:
I do wonder if we should return something richer than a raw PropertyMap here, and have the implementation of `readStackResourceOutputs` in the built in provider translate that into a meaninful property map.  Right now the shape of the final type we are going to send over the wire is split between this method and that one, but I think it would be cleaner if we didn't return raw property maps until much later.  If we just returned a map[URN]struct{tokens.Type, resource.PropertyMap} here, does that make things overall cleaner?

Reply:
Let's do it this way. I don't think it makes it much clearer, and this way the signature differs from `GetStackOutputs` (which is what this is modeled after) but I also don't think it's bad enough to dissent on.

Reply:
@lukehoban @ellismg actually, the whole stack ref reading routines are implemented as we implement this new stuff. I'd prefer to keep them consistent. Objections?